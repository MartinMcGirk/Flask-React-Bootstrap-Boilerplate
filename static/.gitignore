node_modules/
.idea/
yarn-error.log
public/dist/
.env.test
.env.development