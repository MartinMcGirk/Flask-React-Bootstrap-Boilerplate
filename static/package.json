{
  "name": "boilerplate-react-app",
  "version": "1.0.0",
  "main": "index.js",
  "author": "Martin McGirk",
  "license": "MIT",
  "scripts": {
    "build:dev": "webpack",
    "build:prod": "webpack -p --env production",
    "dev-server": "webpack-dev-server",
    "test": "cross-env NODE_ENV=test jest --config=jest.config.json",
    "test:watch": "cross-env NODE_ENV=test jest --config=jest.config.json --watch",
    "start": "node server/server.js",
    "heroku-postbuild": "yarn run build:prod"
  },
  "dependencies": {
    "@fortawesome/fontawesome": "^1.1.8",
    "@fortawesome/fontawesome-free-regular": "^5.0.13",
    "@fortawesome/fontawesome-free-solid": "^5.0.13",
    "@fortawesome/react-fontawesome": "^0.0.20",
    "axios": "^0.18.0",
    "babel-cli": "6.24.1",
    "babel-core": "6.25.0",
    "babel-loader": "7.1.1",
    "babel-plugin-transform-class-properties": "6.24.1",
    "babel-plugin-transform-object-rest-spread": "^6.26.0",
    "babel-polyfill": "6.26.0",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "bootstrap": "^4.1.1",
    "css-loader": "0.28.4",
    "express": "4.15.4",
    "extract-text-webpack-plugin": "3.0.0",
    "firebase": "4.2.0",
    "history": "4.7.2",
    "jquery": "^3.3.1",
    "jwt-decode": "^2.2.0",
    "moment": "2.18.1",
    "node-sass": "4.5.3",
    "normalize.css": "7.0.0",
    "numeral": "2.0.6",
    "popper.js": "^1.14.3",
    "raf": "3.3.2",
    "react": "16.0.0",
    "react-addons-shallow-compare": "15.6.0",
    "react-data-grid": "^4.0.7",
    "react-dates": "12.7.0",
    "react-dom": "16.0.0",
    "react-modal": "2.2.2",
    "react-redux": "5.0.5",
    "react-router-dom": "4.2.2",
    "react-tag-input": "^5.0.2",
    "redux": "3.7.2",
    "redux-mock-store": "1.2.3",
    "redux-thunk": "2.2.0",
    "sass-loader": "6.0.6",
    "style-loader": "0.18.2",
    "uuid": "3.1.0",
    "validator": "^9.4.1",
    "webpack": "3.1.0"
  },
  "devDependencies": {
    "cross-env": "5.0.5",
    "dotenv": "4.0.0",
    "enzyme": "3.0.0",
    "enzyme-adapter-react-16": "1.0.0",
    "enzyme-to-json": "3.0.1",
    "file-loader": "^1.1.11",
    "image-webpack-loader": "^4.2.0",
    "jest": "20.0.4",
    "react-test-renderer": "16.0.0",
    "webpack-dev-server": "2.5.1"
  }
}
