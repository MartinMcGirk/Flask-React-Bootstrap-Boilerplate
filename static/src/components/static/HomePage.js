import React, { Component } from 'react'
import { Link } from 'react-router-dom';
import FontAwesomeIcon from '@fortawesome/react-fontawesome'
import { faSearchPlus, faUserTag, faClock } from '@fortawesome/fontawesome-free-solid'
import { history } from '../../routers/AppRouter';

class HomePage extends Component {
    state = {
        email: ''
    };

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ({
            email
        }));
    };

    onRegisterClick = (e) => {
        e.preventDefault();
        history.push({
            pathname: '/register',
            state: { email: this.state.email }
        });
    };

    render() {
        return (
            <div>
                <header className="masthead text-white text-center">
                    <div className="overlay">

                    </div>
                    <div className="container">
                        <div className="row">
                            <div className="col-xl-9 mx-auto">
                                <h1 className="">This is a boilerplate app</h1>
                                <p>It uses Flask, React, Redux, and Bootstrap</p>
                            </div>
                            <div className="col-md-10 col-lg-8 col-xl-7 mx-auto">
                                <form onSubmit={this.onRegisterClick}>
                                    <div className="form-row">
                                        <div className="col-12 col-md-9 mb-2 mb-md-0">
                                            <input
                                                type="email"
                                                className="form-control form-control-lg"
                                                placeholder="Enter your email..."
                                                value={this.state.email}
                                                onChange={this.onEmailChange}
                                            />
                                        </div>
                                        <div className="col-12 col-md-3">
                                            <button
                                                type="submit"
                                                className="btn btn-block btn-lg btn-primary"
                                            >Sign up!</button>
                                        </div>
                                    </div>
                                </form>
                            </div>
                        </div>
                    </div>
                </header>
                <section className="features-icons bg-light text-center">
                    <div className="container">
                        <div className="row">
                            <div className="col-lg-4">
                                <div className="features-icons-item mx-auto mb-0 mb-lg-3">
                                    <div className="features-icons-icon d-flex">
                                        <FontAwesomeIcon className="m-auto text-primary" icon={faClock}/>
                                    </div>
                                    <h3>Uses postgres</h3>
                                    <p className="lead mb-0">You need to bring your own postgres database.</p>
                                </div>
                            </div>
                            <div className="col-lg-4">
                                <div className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3">
                                    <div className="features-icons-icon d-flex">
                                        <FontAwesomeIcon className="m-auto text-primary" icon={faSearchPlus}/>
                                    </div>
                                    <h3>Develop quickly</h3>
                                    <p className="lead mb-0">You can run the flask and frontend parts simultaneously, with hot reloading when you change things.</p>
                                </div>
                            </div>
                            <div className="col-lg-4">
                                <div className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3">
                                    <div className="features-icons-icon d-flex">
                                        <FontAwesomeIcon className="m-auto text-primary" icon={faUserTag}/>
                                    </div>
                                    <h3>Deploys nicely to heroku</h3>
                                    <p className="lead mb-0">Theoretically this should be easy peasy to deploy to heroku</p>
                                </div>
                            </div>

                        </div>
                    </div>
                </section>
            </div>
        );
    }
}

export { HomePage }