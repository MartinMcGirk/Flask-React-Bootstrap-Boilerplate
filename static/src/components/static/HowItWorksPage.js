import React from 'react'
import { Link } from 'react-router-dom';

const HowItWorksPage = (props) => (
    <div className="container-fluid">
        This is where we will explain how the site works
    </div>
);

export { HowItWorksPage }