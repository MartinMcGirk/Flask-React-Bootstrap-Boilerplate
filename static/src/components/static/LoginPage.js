import React, { Component } from 'react'
import { startLogin } from '../../actions/auth';
import { connect } from 'react-redux';


class LoginPage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: ''
        }
    }

    onUsernameChange = (e) => {
        const username = e.target.value;
        this.setState(() => ({
            username
        }));
    };

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({
            password
        }));
    };

    onLoginSubmit = (e) => {
        e.preventDefault();
        this.props.startLogin(this.state.username, this.state.password);
    };

    render() {
        return (
            <div className="container">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">Boilerplate</h1>
                    <p>Log in!</p>
                </div>
                <form>
                    <div className="form-group">
                        <label htmlFor="exampleInputEmail1">Email address</label>
                        <input
                            type="email"
                            className="form-control"
                            id="exampleInputEmail1"
                            aria-describedby="emailHelp"
                            placeholder="Enter email"
                            value={this.state.username}
                            onChange={this.onUsernameChange}
                        />
                        <small id="emailHelp" className="form-text text-muted">We'll never share your email with anyone else.</small>
                    </div>
                    <div className="form-group">
                        <label htmlFor="exampleInputPassword1">Password</label>
                        <input
                            type="password"
                            className="form-control"
                            id="exampleInputPassword1"
                            placeholder="Password"
                            value={this.state.password}
                            onChange={this.onPasswordChange}
                        />
                    </div>
                    <button type="submit" className="btn btn-primary" onClick={this.onLoginSubmit}>Submit</button>
                </form>
            </div>
        )
    }
}


const mapDispatchToProps = (dispatch) => ({
    startLogin: (username, password) => dispatch(startLogin(username, password))
});

const ConnectedLoginPage = connect(undefined, mapDispatchToProps)(LoginPage);

export { LoginPage, ConnectedLoginPage }