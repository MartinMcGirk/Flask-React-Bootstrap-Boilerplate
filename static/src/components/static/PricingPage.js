import React from 'react'
import { Link } from 'react-router-dom';

const PricingPage = (props) => (
    <div className="container-fluid">
        This will be the pricing page
    </div>
);

export { PricingPage }