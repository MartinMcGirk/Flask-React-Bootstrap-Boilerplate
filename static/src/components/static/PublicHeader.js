import React from 'react'
import { Link } from 'react-router-dom';

const PublicHeader = (props) => (
    <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
        <Link className="navbar-brand mb-0 h1" to="/">
            Boilerplate
        </Link>
        <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
        >
            <span className="navbar-toggler-icon"/>
        </button>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <ul className="navbar-nav mr-auto">
                <li className="nav-item">
                    <Link className="nav-link" to="/about">
                        How it works
                    </Link>
                </li>
                <li className="nav-item">
                    <Link className="nav-link" to="/pricing">
                        Pricing
                    </Link>
                </li>
            </ul>
            <ul className="navbar-nav">
                <li className="nav-item">
                    <Link className="btn btn-outline-light" to="/login">Login</Link>
                </li>
                <li className="nav-item">
                    <Link className="btn btn-outline-light" to="/register">Register</Link>
                </li>
            </ul>
        </div>
    </nav>
);

export { PublicHeader }