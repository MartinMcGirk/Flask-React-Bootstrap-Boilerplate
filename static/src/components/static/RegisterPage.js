import React, { Component } from 'react'
import { FormInput } from '../FormInput';
import FontAwesomeIcon from '@fortawesome/react-fontawesome'
import { faCheckCircle as fasCircle } from '@fortawesome/fontawesome-free-solid'
import { faCircle as farCircle } from '@fortawesome/fontawesome-free-regular'
import { connect } from 'react-redux';
import { startRegisterUser } from '../../actions/auth';

class RegisterPageComponent extends Component {
    constructor(props) {
        super(props);
        console.log(props);
        let email = '';
        if (props.location.state) {
            email = props.location.state.email || '';
        }
        this.state = {
            email,
            password: '',
            firstName: '',
            lastName: '',
            role: 'candidate'
        }
    }

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ({ email }));
    };

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    };

    onFirstNameChange = (e) => {
        const firstName = e.target.value;
        this.setState(() => ({ firstName }));
    };

    onLastNameChange = (e) => {
        const lastName = e.target.value;
        this.setState(() => ({ lastName }));
    };

    // onRoleChange = (e) => {
    //     e.preventDefault();
    //     const role = e.target.value;
    //     this.setState(() => ({ role }));
    // };

    onRegisterFormSubmit = (e) => {
        e.preventDefault();

        if (this.state.email && this.state.password
        && this.state.firstName && this.state.lastName
        && this.state.role) {
            this.props.startRegisterUser({ ...this.state });
        } else {
            alert('You need to fill in all the fields');
        }

    };

    render() {
        return (
            <div className="container">
                <div className="row justify-content-center">
                    <div className="col col-lg-6 col-md-8 col-sm-12 align-self-center">
                        <h2 className="page-header">Join Boilerplate today</h2>
                    </div>
                </div>

                <div className="row justify-content-center">
                    <div className="col col-lg-6 col-md-8 col-sm-12 align-self-center">
                        <form onSubmit={this.onRegisterFormSubmit}>
                            <FormInput
                                fieldId="email"
                                inputType="email"
                                label="Email"
                                placeholder="Your email"
                                smallText="We'll never share your email with anyone else."
                                value={this.state.email}
                                onFieldChange={this.onEmailChange}
                            />
                            <FormInput
                                fieldId="password"
                                inputType="password"
                                label="Password"
                                placeholder="Password"
                                value={this.state.password}
                                onFieldChange={this.onPasswordChange}
                            />
                            <FormInput
                                fieldId="firstName"
                                label="First Name"
                                placeholder="Your first name"
                                value={this.state.firstName}
                                onFieldChange={this.onFirstNameChange}
                            />
                            <FormInput
                                fieldId="lastName"
                                label="Last Name"
                                placeholder="Your last name"
                                value={this.state.lastName}
                                onFieldChange={this.onLastNameChange}
                            />
                            {/*<div className="form-row">*/}
                                {/*<div className="col-md-5 col-sm-12">*/}
                                    {/*<button*/}
                                        {/*className="btn btn-outline-primary btn-block"*/}
                                        {/*value="candidate"*/}
                                        {/*onClick={this.onRoleChange}*/}
                                    {/*>*/}
                                        {/*<FontAwesomeIcon className="pr-1" icon={this.state.role === 'candidate' ? fasCircle : farCircle} />*/}
                                        {/*I want to work*/}
                                    {/*</button>*/}
                                {/*</div>*/}
                                {/*<div className="col text-center">*/}
                                    {/*OR*/}
                                {/*</div>*/}
                                {/*<div className="col-md-5 col-sm-12">*/}
                                    {/*<button*/}
                                        {/*className="btn btn-outline-primary btn-block"*/}
                                        {/*value="recruiter"*/}
                                        {/*onClick={this.onRoleChange}*/}
                                    {/*>*/}
                                        {/*<FontAwesomeIcon className="pr-1" icon={this.state.role === 'recruiter' ? fasCircle : farCircle} />*/}
                                        {/*I want to hire*/}

                                    {/*</button>*/}
                                {/*</div>*/}
                            {/*</div>*/}
                            <div className="form-row">
                                <div className="col mt-3">
                                    <button className="btn btn-primary btn-lg btn-block">
                                        Get started
                                    </button>
                                </div>
                            </div>
                        </form>
                    </div>
                </div>

            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startRegisterUser: (userDetails) => dispatch(startRegisterUser(userDetails))
});

const RegisterPage = connect(null, mapDispatchToProps)(RegisterPageComponent);

export { RegisterPage }