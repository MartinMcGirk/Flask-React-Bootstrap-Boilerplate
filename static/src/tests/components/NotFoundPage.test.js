import React from 'react';
import { shallow } from 'enzyme';
import { NotFoundPage } from '../../components/static/NotFoundPage';

test('should correctly render NotFoundPage', () => {
    const wrapper = shallow(<NotFoundPage/>);
    expect(wrapper).toMatchSnapshot();
});
