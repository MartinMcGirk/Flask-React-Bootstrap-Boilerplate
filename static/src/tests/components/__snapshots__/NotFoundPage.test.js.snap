// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render NotFoundPage 1`] = `
<div>
  404! - 
  <Link
    replace={false}
    to="/"
  >
    Go home
  </Link>
</div>
`;
